src/libstore/build/derivation-goal.cc

```diff
@@ -985,21 +985,29 @@ void DerivationGoal::resolvedFinished()
             realWantedOutputs = resolvedDrv.outputNames();
 
         for (auto & wantedOutput : realWantedOutputs) {
-            assert(initialOutputs.count(wantedOutput) != 0);
-            assert(resolvedHashes.count(wantedOutput) != 0);
-            auto realisation = resolvedResult.builtOutputs.at(
-                DrvOutput { resolvedHashes.at(wantedOutput), wantedOutput });
+            auto initialOutput = initialOutputs.find(wantedOutput);
+            auto resolvedHash = resolvedHashes.find(wantedOutput);
+            if ((initialOutputs.end() == initialOutput) || (resolvedHashes.end() == resolvedHash))
+                throw Error(
+                    "derivation '%s' doesn't have expected output '%s' (derivation-goal.cc/resolvedFinished,resolve)",
+                    worker.store.printStorePath(drvPath), wantedOutput);
+            auto realisation = resolvedResult.builtOutputs.find(
+                DrvOutput { resolvedHash->second, wantedOutput });
+            if (resolvedResult.builtOutputs.end() == realisation)
+                throw Error(
+                    "derivation '%s' doesn't have expected output '%s' (derivation-goal.cc/resolvedFinished,realisation)",
+                    worker.store.printStorePath(resolvedDrvGoal->drvPath), wantedOutput);
```

Comment:
This is the exception hit in #6383.

Reply:
maybe `resolvedHash->second` mismatches? Is there some way to format `resolvedResult.builtOutputs` to include it in the error message?